### PR TITLE
Ajax-ify protection message (inherited ACL)

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: indico/modules/events/contributions/controllers/management.py [hunk 1]
```diff
@@ -227,6 +227,16 @@ def _process(self):
         return render_acl(self.contrib)
 
 
+class RHContributionACLMessage(RHManageContributionBase):
+    """Render the inheriting ACL message"""
+
+    def _process(self):
+        inheriting = request.args.get('inheriting')
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/controllers/management.py]: I'd go for a more simple `request.args.get('inheriting') == '1'` here.

File: indico/web/templates/_inherited_acl_message_base.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{% block message %}
+<div class="action-box for-form highlight inheriting-acl-message">
+    <div class="section">
+        <div class="text">
+            <div>
+                {% trans %}Since the protection mode is set to <em>inheriting</em>, users from the <strong>inherited access control list</strong> will also have access.{% endtrans %}
```
Comment (ThiefMaster) [inline indico/web/templates/_inherited_acl_message_base.html]: what about some linebreaks?

File: indico/web/templates/forms/protection_widget.html [hunk 3]
```diff
@@ -23,7 +23,14 @@
 
                     toggleAclField($('#{{ field.get_form().acl.id }}'), !isProtected);
 
-                    $this.closest('form').find('.inheriting-acl-message').trigger('protection:inheriting', [$this.val() === 'inheriting']);
+                    $.ajax({
+                        url: '{{ field.acl_message_url }}',
```
Comment (ThiefMaster) [inline indico/web/templates/forms/protection_widget.html]: `|tojson` instead of quotes

File: indico/web/forms/fields.py [hunk 4]
```diff
@@ -844,6 +838,8 @@ class IndicoProtectionField(IndicoEnumRadioField):
     radio_widget = JinjaWidget('forms/radio_buttons_widget.html', orientation='horizontal', single_kwargs=True)
 
     def __init__(self, *args, **kwargs):
+        self.protected_object = kwargs.pop('protected_object')(kwargs['_form'])
+        self.acl_message_url = kwargs.pop('acl_message_url')(kwargs['_form'])
```
Comment (ThiefMaster) [inline indico/web/forms/fields.py]: I would make this optional.

Comment (nop33) [inline indico/web/forms/fields.py]: Are we really gonna have a **protection mode** field without an **Access control list
** field?

Comment (ThiefMaster) [inline indico/web/forms/fields.py]: Maybe not without an ACL, but do we **always** have the ajaxified message? WHat about toplevel categories  with no parent?

**Pull request merged**